packages/indexer/src/starknet/parser.ts

```diff
@@ -0,0 +1,56 @@
+import { FieldElement } from "./felt";
+import { EventFilter } from "./filter";
+import { EventWithTransaction } from "./block";
+
+import { type Abi, hash } from "starknet";
+
+interface EventFilterParser {
+  eventFilter(name: string): EventFilter;
+  eventParser(name: string): ({ event, transaction }: EventWithTransaction ) => any;
+}
+
+// The Contract class implements the EventFilterParser interface.
+export class Contract implements EventFilterParser {
+  // Read-only properties for the contract's address and its ABI.
+  readonly contractAddress: FieldElement;
+  readonly contractAbi: Abi;
+
+  // Constructor to initialize a Contract instance with a contract address and ABI.
+  constructor(contractAddress: FieldElement, contractAbi: Abi) {
+    this.contractAddress = contractAddress;
+    this.contractAbi = contractAbi;
+  }
+
+  // Method to filter events based on their name from the contract's ABI.
+  eventFilter(name: string): EventFilter {
+    // Find the event in the ABI matching the provided name.
+    const event: EventFilter = this.contractAbi.find((item) => item.name === name);
+    
+    // Throw an error if the event is not found in the ABI
+    if (!event) {
+      throw new Error(`Event ${name} not found in contract ABI`);
+    }
+
+    // Return the found event.
+    return event;
```

Comment:
Here we want to build an `EventFilter` starting from the `EventAbi` above. For now, let's do something like the following.
```suggestion
    // Return the found event.
    return {
      fromAddress: this.contractAddress,
      keys: [hash.getSelectorFromName(event.name)],
      includeTransaction: false,
      includeReceipt: false,
    };
```

Reply:
The event filter used here can be used in the indexer's `config.filter`.

```ts
const contract = new Contract(ETH_ADDRESS, ETH_ABI);

export const config = {
  // ... other config here...
  filter: {
    header: { weak: true },
    event: [
      contract.eventFilter("Approve"),
      contract.eventFilter("Transfer"),
    ]
  }
}
```